### PR TITLE
New conversation stories & tests

### PR DESCRIPTION
**Pull request by AmeeMorris**

### CODE REVIEW

File: lib/Conversation/Comment.js [hunk 1]
```diff
@@ -14,19 +14,15 @@ class Comment extends Component {
...
   };
 
   static defaultProps = {
     user: {},
     focusFallback() {},
-    actions: {
-      editComment() {},
-      removeComment() {}
-    }
+    editComment() {},
```
Comment (kyleharper) [inline lib/Conversation/Comment.js]: Remove as these are required 👍

File: lib/Conversation/index.js [hunk 2]
```diff
@@ -4,18 +4,16 @@ import pluralize from 'pluralize';
...
-    actions: PropTypes.shape({
-      resolveConversation: PropTypes.func.isRequired,
-      addComment: PropTypes.func.isRequired,
-      activateConversation: PropTypes.func.isRequired,
-      deactivateConversation: PropTypes.func.isRequired
-    }).isRequired,
+    resolveConversation: PropTypes.func,
+    onCancel: PropTypes.func,
+    addComment: PropTypes.func.isRequired,
+    activateConversation: PropTypes.func,
```
Comment (kyleharper) [inline lib/Conversation/index.js]: Lets abstract out `activateConversation` and `deactivateConversation` and treat BoundaryClickWatcher as another UI component that can be used to compose the interactions we desire.

File: lib/Conversation/index.js [hunk 3]
```diff
@@ -37,85 +40,85 @@ class Conversation extends Component {
...
-            conversationId={id}
-            comments={comments}
-            focusFallback={this.focusFallback}
-          />
-
-          {!isActive &&
-            comments.length > 1 && (
-              <div className="conversation__reply-count">
+      <div className={`conversation ${isActiveClass}`}>
+        {typeof this.props.resolveConversation !== 'undefined' &&
```
Comment (kyleharper) [inline lib/Conversation/index.js]: Try just `!this.props.resolveConversation` 👍

File: tests/Conversation/Conversation.js [hunk 4]
```diff
@@ -160,4 +150,9 @@ describe('Conversation', () => {
     wrapper.setProps({ userCanComment: false });
     expect(wrapper.find(CommentForm)).to.have.length(0);
   });
+
+  it('does not render a resolveConversation button', () => {
```
Comment (kyleharper) [inline tests/Conversation/Conversation.js]: We also don't render a convoList 😄

File: lib/Conversation/index.js [hunk 5]
```diff
@@ -24,98 +20,89 @@ class Conversation extends Component {
...
-            type="text"
-            className="conversation__focus-fallback"
-            ref={focusFallback => {
-              this.focusFallback = focusFallback;
-            }}
-            aria-hidden="true"
-          />
-        </div>
-      </BoundaryClickWatcher>
+        {!isActive &&
```
Comment (kyleharper) [inline lib/Conversation/index.js]: I wonder if we should change the naming of `isActive` to something like `showComments`. The original naming made sense as we were setting the active state internally but now we're just showing the comments or not. What do you think?

**Comment by AmeeMorris** [comment]
Hey @kyleharper I've also added in the BoundaryClickWatcher refactoring to this PR

File: lib/BoundaryClickWatcher/index.js [hunk 6]
```diff
@@ -67,10 +75,13 @@ class BoundaryClickWatcher extends Component {
...
         }}
-        onClick={this.handleInitialClick}
-        onKeyPress={this.handleInitialClick}
+        onClick={this.handleInnerClick}
+        onKeyPress={this.handleInnerClick}
       >
-        {this.props.children}
+        {typeof this.props.children === 'function'
+          ? this.props.children(this.state.boundaryIsActive)
+          : this.props.children
```
Comment (kyleharper) [inline lib/BoundaryClickWatcher/index.js]: Is this so we can use this component as a means to trigger `initialClickHandler` & `outsideClickHandler` without passing the active state to the child?

File: tests/BoundaryClickWatcher/BoundaryClickWatcher.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    );
+  });
+
+  it('boundaryIsActive is true on click', () => {
+    expect(wrapper.state('boundaryIsActive')).to.equal(false);
+    wrapper.simulate('click');
+    expect(wrapper.state('boundaryIsActive')).to.equal(true);
+  });
+
+  it('passes boundaryIsActive to child', () => {
```
Comment (kyleharper) [inline tests/BoundaryClickWatcher/BoundaryClickWatcher.js]: Did you want to test when a child is passed that isn't a function? If you can't imagine a use case for this then maybe we should expect a function to be passed as the child 👍

File: lib/BoundaryClickWatcher/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+##Usage
+
+BoundaryClickWatcher has a boolean (boundaryIsActive) which is set to true when it detects an inner click and back to false when it detects a click outside of the boundary. The value of this can be passed onto the child component like so:
+
+```
+<BoundaryClickWatcher>
+  {(boundaryIsActive) => {
+    return (
+      <My Component
```
Comment (kyleharper) [inline lib/BoundaryClickWatcher/README.md]: Change to `<MyComponent` 😄

File: lib/BoundaryClickWatcher/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
+#BoundaryClickWatcher
+Wrap a component in BoundaryClickWatcher to detect clicks on our outside of the component. See an example of this in the Conversation story.
+
+##Usage
+
+BoundaryClickWatcher has a boolean (boundaryIsActive) which is set to true when it detects an inner click and back to false when it detects a click outside of the boundary. The value of this can be passed onto the child component like so:
```
Comment (kyleharper) [inline lib/BoundaryClickWatcher/README.md]: I'd say its worth mentioning that it uses the render prop pattern and keep this simple. Something like: "`BoundaryClickWatcher` uses the render callback pattern and returns a boolean."

I think the usage shows just how to use it 😃

File: lib/BoundaryClickWatcher/README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+      <My Component
+        boolProp={boundaryIsActive}
+      />
+    )
+  }}
+</BoundaryClickWatcher>
+```
+
+*(Optional)* It has two props `insideClickHandler` and `outsideClickHandler` you can pass in actions for added custom behaviour on inner or outer click.
```
Comment (kyleharper) [inline lib/BoundaryClickWatcher/README.md]: I wonder if we should make this more concise?

Something like the following section in a different library: https://github.com/igorprado/react-notification-system#methods

e.g.

--------

## Props

### `insideClickHandler`

Executes when the user clicks inside the boundary.

### ```outsideClickHandler```

Executes when the user clicks outside the boundary.

--------

Or maybe a table like https://github.com/igorprado/react-notification-system/blob/master/README.md#creating-a-notification

What do you think?

File: lib/Conversation/README.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
+#Conversation
+A collection of components used to render conversations
+
+##Usage
+
+####Required props
```
Comment (kyleharper) [inline lib/Conversation/README.md]: I think a table would work better for props, what do you think?

**Pull request merged**